### DISCUSSION TITLE
New Mercenary System (ALE)

### DISCUSSION THREAD

**Discussion by cressidagp** [arcemu/arcemu#464] [Show and tell]
Hi there sweet community!

Back at 2019 (while covid lockdown), i started to work on a new Mercenary System for ArcEmu. Since then, i did many versions of this, theres one where mercenarys start as civilian recruit and then can evolve in to different class, there was another version where mercs are obtained by quests. Over the years, restarted the project several times. Recently, i started to feel confident with the new path taked so decided to show this experimental version.

Features:

- Mercenary Masters on each capital
- Hire up to three Soldiers
- Mercs will have same level than player
- They will eat, drink, emote when gets bored
- Three fight styles for Soldiers: balanced, defensive, offensive
- In balanced style they will charge when enter combat
- Race, gender and display customizable in game (this may change due cache issues)
- Spell progresion: up to nine ranks for each spell
- Database storage: mercs will be loaded on every map change

ToDo:

- Hire abandoned mercs on the field
- Soldiers: spells for defensive and offensive style
- Medic 
- Officer
- Cleanup all scripts
- Make them more usefull for players who do Raids alone
- Find a way (if there is one) to move our mercs like an RTS

Vendor pic:

![WoWScrnShot_123024_151154](https://github.com/user-attachments/assets/5e737234-27fb-4407-a6ba-497a9f4fa376)

Interface pic:

![WoWScrnShot_123024_150737](https://github.com/user-attachments/assets/857f5047-4ffc-450d-bf84-751de50a0f4b)

Lua Scripts:

- masterHandler: https://pastebin.com/eBRxyW4d
- storageHandler: https://pastebin.com/yG4vwLZB
- interfaceHandler: https://pastebin.com/qLQbHTF6
- ia_soldier: https://pastebin.com/YEkRqhmd

Querys for World Database: https://pastebin.com/U55kNzBz

Arcmercs Database: https://pastebin.com/gP7LiYME

Credits:

- Hypersniper and Halestorm (original idea)
- dfighter1985 for develop this kickass emu and keep us well provided of scripting engine(s)

**Comment by dfighter1985** [comment]
So If I understand correctly with this basically you get an NPC or more NPCs who helps you while playing the game.
Sounds interesting!

**Reply by cressidagp** [reply]
That is correct Great One. There will be a new version soon... :)
